src/Pact/Analyze/Check.hs

```diff
@@ -1028,7 +1036,7 @@ getStepChecks env@(CheckEnv tables consts propDefs _ _ _) defpactRefs = do
     Left errs         -> throwError $ ModuleParseFailure errs
     Right stepChecks' -> pure stepChecks'
 
-  lift $ ifor stepChecks' $
+  lift $ fmap (fmap (nub . concat)) $ ifor stepChecks' $
```

Comment:
Is this `nub` necessary? I would assume the step checks wouldn't overlap by construction

Reply:
The warnings do, especially the unscoped cap warning. While each check uses a `Set` the results definitely have dupes.

Reply:
Why not collect a `Set` of warnings instead of a list then?

Reply:
Mainly because the interface already exposes a list, so going in and out of Set is annoying when we can just `nub`. These lists are small

Reply:
There's potentially a declaration-order thing going on too, less important to preserve but Set would destroy it